.github/scripts/create-flaky-test-report.ts

```diff
@@ -0,0 +1,272 @@
+              date: new Date(testSuite.date),
+            }))
+            .concat(
+              testSuite.attempts.flatMap((attempt) =>
+                attempt.testCases.map((testCase) => ({
+                  ...testCase,
+                  path: testFile.path,
+                  jobId: attempt.job.id,
+                  runId: attempt.job.runId,
+                  date: new Date(attempt.date),
+                })),
+              ),
+            )
+            .filter((testCase) => testCase.status === 'failed'),
+        ),
+      ),
+    )
+    .sort((a, b) => b.date.getTime() - a.date.getTime());
+
+  const summarizedFailedTests = Object.values(
+    failedTests.reduce<
+      Record<string, (typeof failedTests)[number] & { count: number }>
+    >((summary, test) => {
+      if (hasProperty(summary, test.name) && summary[test.name]) {
+        summary[test.name].count += 1;
+      } else {
+        summary[test.name] = { ...test, count: 1 };
+      }
+      return summary;
+    }, {}),
+  ).sort((a, b) => b.count - a.count);
+
+  const options = { year: 'numeric', month: 'long', day: 'numeric' } as const;
+  const fromDateString = from.toLocaleDateString('en-US', options);
+  const toDateString = to.toLocaleDateString('en-US', options);
+
+  const repositoryUrl = new URL(
+    `https://github.com/${env.OWNER}/${env.REPOSITORY}`,
+  );
+  const branchUrl = new URL(`${repositoryUrl}/tree/${env.BRANCH}`);
+
+  console.log(
+    `❌ Failed tests on the ${env.REPOSITORY} repository ${env.BRANCH} branch from ${fromDateString} to ${toDateString}`,
+  );
+  const blocks: AnyBlock[] = [
+    {
+      type: 'rich_text',
+      elements: [
+        {
+          type: 'rich_text_section',
+          elements: [
+            {
+              type: 'emoji',
+              name: 'x',
+            },
+            {
+              type: 'text',
+              text: ' Failed tests on the ',
+            },
+            {
+              type: 'link',
+              url: repositoryUrl.toString(),
+              text: env.REPOSITORY,
+            },
+            {
+              type: 'text',
+              text: ' repository ',
+            },
+            {
+              type: 'link',
+              url: branchUrl.toString(),
+              text: env.BRANCH,
+            },
+            {
+              type: 'text',
+              text: ` branch from ${fromDateString} to ${toDateString} on GitHub Actions`,
+            },
+          ],
+        },
+      ],
+    },
+  ];
+
+  if (!summarizedFailedTests.length) {
+    console.log('No failed tests found, good job!');
+    blocks.push({
+      type: 'rich_text',
+      elements: [
+        {
+          type: 'rich_text_section',
+          elements: [
+            {
+              type: 'text',
+              text: 'No failed tests found, good job!',
+            },
+          ],
+        },
+      ],
+    });
+  } else {
+    for (const test of summarizedFailedTests) {
+      const testUrl = new URL(
```

Comment:
if `test.path` has any special url characters they will be treated as such. they need to be escaped.